From 54c86ca6424b6ccd3e970fc80d3a45f93d07dceb Mon Sep 17 00:00:00 2001
From: Dong Aisheng <b29396@freescale.com>
Date: Wed, 6 Nov 2013 10:50:09 +0800
Subject: [PATCH] ENGR00286724-7 ARM: dts: sabreauto: add flexcan support

The flexcan1 is disabled by default since it's conflict with fec.

Signed-off-by: Dong Aisheng <b29396@freescale.com>
(cherry picked from commit 0f6cfd8708633ee2f924aa509b30bb17b185f84e)
---
 arch/arm/boot/dts/imx6qdl-sabreauto.dtsi | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
index 0a4ffc5..84a39de 100644
--- a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
@@ -179,6 +179,20 @@
 			>;
 		};
 
+		pinctrl_flexcan1: flexcan1grp {
+			fsl,pins = <
+				MX6QDL_PAD_KEY_ROW2__FLEXCAN1_RX	0x80000000
+				MX6QDL_PAD_KEY_COL2__FLEXCAN1_TX	0x80000000
+			>;
+		};
+
+		pinctrl_flexcan2: flexcan2grp {
+			fsl,pins = <
+				MX6QDL_PAD_KEY_COL4__FLEXCAN2_TX	0x80000000
+				MX6QDL_PAD_KEY_ROW4__FLEXCAN2_RX	0x80000000
+			>;
+		};
+
 		pinctrl_gpmi_nand: gpminandgrp {
 			fsl,pins = <
 				MX6QDL_PAD_NANDF_CLE__NAND_CLE		0xb0b1
@@ -426,6 +440,23 @@
 	status = "okay";
 };
 
+&flexcan1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_flexcan1>;
+	pinctrl-assert-gpios = <&max7310_b 3 GPIO_ACTIVE_HIGH>; /* TX */
+	trx-en-gpio = <&max7310_b 6 GPIO_ACTIVE_HIGH>;
+	trx-stby-gpio = <&max7310_b 5 GPIO_ACTIVE_HIGH>;
+	status = "disabled"; /* pin conflict with fec */
+};
+
+&flexcan2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_flexcan2>;
+	trx-en-gpio = <&max7310_c 6 GPIO_ACTIVE_HIGH>;
+	trx-stby-gpio = <&max7310_c 5 GPIO_ACTIVE_HIGH>;
+	status = "okay";
+};
+
 &hdmi_audio {
 	status = "okay";
 };
-- 
1.8.0

